### PR TITLE
make parameter a real struct

### PR DESCRIPTION
**Pull request by anandijain**
@exaexa Locally all tests are passing except something in flux_bounds fails. It should be an obvious change.
The reason I need this is I have a model with units in params.

### CODE REVIEW

File: src/readsbml.jl [hunk 1]
```diff
@@ -147,7 +147,7 @@ end)
 function readSBML(
     fn::String,
     sbml_conversion = document -> nothing;
-    report_severities = ["Fatal", "Error"],
+    report_severities = ["Fatal", "Error"]
```
Comment (giordano) [inline src/readsbml.jl:150]: Why removing the commas?  They're useful to add a new argument without disrupting git history, which is exactly what you're doing here.

Comment (anandijain) [reply src/readsbml.jl:150]: i think the vscode auto formatter did that

**Comment by anandijain** [comment]
I will add a test, but unfortunately, I can't find a single sbml-test-suite case that actually uses units in parameters. 
I grepped with  `grep -R -l "<parameter.*unit"` and got nothing. Mose, can you sanity check before I open an issue in the test-suite repo. I feel like they should probably have at least one.

File: src/readsbml.jl (hunk 1 above)
Comment (exaexa) [reply src/readsbml.jl:150]: JuliaFormatter to the git hook! :D

**Comment by exaexa** [comment]
This kinda materializes `Parameter_t`, right?

Looks good to me, except we might stay consistent with the rest of SBML.jl and report units in normal tuples. (not that I'd like to support the technical debt in that part, but it's going to make the transition much easier)

**Comment by anandijain** [comment]
I'm not exactly sure what you mean wrt the last part of tuples. What's the best way to go about that?
I could define `convert(Tuple, Parameter)` right? seems weird

**Comment by exaexa** [comment]
> I'm not exactly sure what you mean wrt the last part of tuples. What's the best way to go about that?

The rest of SBML.jl represents all values in tuples of `(x::Real,unit::String)`; if you maintain that convention then both working with the current code _and_ the later migration to something better will be easier.

**Comment by anandijain** [comment]
I see, what is a good way to do that though? This latest commit demonstrates my confusion haha

**Comment by anandijain** [comment]
okay this should pass but I'm sure you have a way to clean it up

**Comment by exaexa** [comment]
I have added a few commits, will merge this branch as a part of #182 . Thanks! (also feel free to review there :D )

**Pull request closed without merging**